transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ScheduleTransferActivity.java

```diff
@@ -0,0 +1,156 @@
+
+    private ScheduleTransferViewModel mScheduleTransferViewModel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_schedule_transfer);
+
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+        getSupportActionBar().setTitle(R.string.title_activity_create_transfer);
+        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                setResult(Activity.RESULT_OK);
+                finish();
+            }
+        });
+
+        Parcelable transferParcel = getIntent().getParcelableExtra(EXTRA_TRANSFER);
+        Parcelable transferMethodParcel = getIntent().getParcelableExtra(EXTRA_TRANSFER_METHOD);
+        if (transferParcel instanceof Transfer && transferMethodParcel instanceof HyperwalletTransferMethod) {
+            mScheduleTransferViewModel = ViewModelProviders.of(this,
+                    new ScheduleTransferViewModel.ScheduleTransferViewModelFactory(
+                            TransferRepositoryFactory.getInstance().getTransferRepository()))
+                    .get(ScheduleTransferViewModel.class);
+            mScheduleTransferViewModel.setTransfer((Transfer) transferParcel);
+            mScheduleTransferViewModel.setTransferDestination((HyperwalletTransferMethod) transferMethodParcel);
+            registerObservers();
+        } else {
+            throw new IllegalArgumentException("Required extra arguments are invalid for "
+                    + ScheduleTransferActivity.class.getName()
+                    + " extra keys: EXTRA_TRANSFER and EXTRA_TRANSFER_METHOD");
+        }
+
+        if (savedInstanceState == null) {
+            initFragment(ScheduleTransferFragment.newInstance());
+        }
+    }
+
+    @Override
+    public void retry() {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        ScheduleTransferFragment fragment = (ScheduleTransferFragment)
+                fragmentManager.findFragmentById(R.id.schedule_transfer_fragment);
+
+        if (fragment == null) {
+            fragment = ScheduleTransferFragment.newInstance();
+        }
+        fragment.retry();
+    }
+
+    private void registerObservers() {
+        mScheduleTransferViewModel.getTransferStatusTransitionError().observe(this,
+                new Observer<Event<HyperwalletErrors>>() {
+                    @Override
+                    public void onChanged(final Event<HyperwalletErrors> event) {
+                        if (event != null && !event.isContentConsumed()) {
+                            showErrorScheduleTransfer(event.getContent().getErrors());
+                            setResult(Activity.RESULT_OK);
```

Comment:
If we are not going to dispose the activity I think we don't need to call **setResult()**, since we are not finishing the activity, there was some question closing the schedule transfer activity when there's an error, probably this is from the prototype... 

but if we are going to change the flow
*) close schedule transfer activity when there's error to go back to create transfer ui
    then we need to to do the ff:
    *.a) set result to **Activity.RESULT_CANCELED** because commit is actually cancelled after an error
    *.b) finish the activity meaning closing it and showing the create transfer UI

Reply:
ok let's leave it as is for now